1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,346 @@
+PSReadLine module. In an effort to support the most PowerShell users, the presentation of
+history-based predictions will be available to the following versions:
+
+- Windows Powershell 5.1
+- PowerShell 7.0+
+
+Additional plugins require the the extension framework implemented in the PowerShell engine. Support
+for additional plugins is available in the following versions:
+
+- PowerShell 7.1+
+
+History-based predictions preview and ship plan:
+
+- PSReadLine 2.1.0-beta2 currently available for download from PSGallery
+- PSReadLine 2.1.0-GA will ship with PowerShell 7.1
+- PSReadLine 2.1.0-GA will be available for download from PSGallery
+- History based prediction in the **InlineView**
+
+Extension/Plugin support preview and ship plan:
+
+- PSReadLine 2.2.0-beta1 planned availability in October
+- PSReadLine 2.2.0-GA will ship with PowerShell 7.2
+- PSReadLine 2.2.0-GA will be available for download from PSGallery
+- Includes support for **InlineView** and **ListView**
+- Includes support for additional prediction source **Plugin** and **HistoryAndPlugin**
+- Note: Windows PowerShell 5.1 and PowerShell 7.0 will only support prediction source **History**.
+  PowerShell 7.1+ will support prediction sources **History**, **Plugin**, and **HistoryAndPlugin**
+
+## Goals/Non-Goals
+
+ Goals:
+
+- Provide suggestions from PSReadLine history
+- The user will be able to enable and disable Predictive IntelliSense
+- The user will be able to change between InlineView and ListView
+- The user will have keyboard shortcuts to navigate and edit a prediction
+- The user will be able to register additional providers when desired
+- The user may customize the color of a prediction to support accessibility needs
+
+Non-goals:
+
+- Due to the relatively short length of a session compared to your PSReadline history,
+  PowerShell session history is not part of the historical suggestion
+- The extension framework doesn't allow a plugin implementation to call into the Runspace where
+  PSReadLine is running when providing predictive suggestions.
+- VSCode implementation. We agree this is important for a consistent experience and are working to
+  bring this feature in a future release, but the user experience for VS Code is outside the scope of this RFC.
+
+## Specification
+
+The proposal is to add Predictive IntelliSense and an extension model to improve the user
+interactive experience.
+
+### Key Design Considerations Predictive IntelliSense
+
+Predictive IntelliSense displays the most recent match command completion from command history. The
+prediction may change if a better match is found as the user types additional information. The
+predicted command may be accepted, edited, or ignored.
+
+![Prediction - Historical](./media/pred-wlc.png)
+
+### Install and Remove Predictive IntelliSense
+
+Predictive IntelliSense is implemented in both the PowerShell engine and presented through the
```

Comment:
You say that a couple of times, that the engine has to have code changes. What are the nature of the engine changes?

Reply:
The API for the predictors and the subsystem is in Powershell.  Changed to Predictive IntelliSense application programming interface (API) is implemented in the PowerShell
engine and presented through the PSReadLine module.